### PR TITLE
[image_picker] Remove input element after completion

### PR DESCRIPTION
**Pull request by Amir-P**
This PR attempts to remove the input element added to the DOM after image selection is completed.

Fixes https://github.com/flutter/flutter/issues/139442

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart [hunk 1]
```diff
@@ -320,6 +320,9 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     // TODO(dit): Reimplement this with the showPicker() API, https://github.com/flutter/flutter/issues/130365
     element.click();
   }
+
+  /// Removes the file input element
+  void _remove(html.Element element) => _target.children.clear();
```
Comment (ditman) [inline packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart:325]: This `_remove(element)` is not removing element, it's clearing _target.children.

File: packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart [hunk 2]
```diff
@@ -158,7 +158,7 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     ) as html.FileUploadInputElement;
     _injectAndActivate(input);
 
-    return _getSelectedXFiles(input);
+    return _getSelectedXFiles(input).whenComplete(() => _remove(input));
```
Comment (ditman) [inline packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart:161]:
```suggestion
    return _getSelectedXFiles(input).whenComplete(() => input.remove());
```

**Review by ditman — CHANGES_REQUESTED** [review]
Small simplification of the fix.

**Comment by Amir-P** [comment]
Done. At first I wanted to just remove the input like you said but I saw that `_target.children.clear()` is called before injecting input element and thought you prefer that. @ditman

**Comment by Amir-P** [comment]
@ditman could you review the PR please? This bug affects the usage of this plugin in our Web app.

**Comment by ditman** [comment]
> This bug affects the usage of this plugin in our Web app.

How so, @Amir-P?

**Review by ditman — APPROVED** [review]
LGTM, thanks for cleaning this up!

**Comment by ditman** [comment]
Ah, we need two reviewers for this one!

/cc @mdebbar can you PTAL? (I think you commented in the OG issue!)

**Review by mdebbar — APPROVED** [review]
Thanks for the fix!

**Pull request merged**